src/main/kotlin/org/jetbrains/plugins/bsp/server/tasks/CollectProjectDetailsTask.kt

```diff
@@ -197,6 +226,42 @@ public class CollectProjectDetailsTask(project: Project, private val taskId: Any
     }
   }
 
+  private suspend fun addBspFetchedPythonSdks() {
+    bspSyncConsole.startSubtask(taskId, "add-bsp-fetched-python-sdks", "Adding BSP-fetched Python SDKs...")
+    logPerformanceSuspend("add-bsp-fetched-python-sdks") { pythonSdks?.forEach { addPythonSdkIfNeeded(it) } }
+    bspSyncConsole.finishSubtask("add-bsp-fetched-python-sdks", "Adding BSP-fetched Python SDKs done!")
+  }
+
+  private suspend fun addPythonSdkIfNeeded(pythonSdk: PythonSdk) {
+    val allJdks = jdkTable.allJdks.toList()
+    val additionalData = PythonSdkAdditionalData()
+    val virtualFiles = pythonSdk.dependencies.mapNotNull {
```

Comment:
u need to obtain real sources from the dependencies and `it.target.uri` is just a target id not a dependency, so u need to do something like:

`dependencies.flatMap { it.sources }.mapNotNull { URI.create(it) [..]`